### PR TITLE
Add parquet file value for match_pattern

### LINKED ISSUE #2371: Document: feat(connector): introduce parquet file source
This issue tracks the documentation update needed for the merged PR #17201.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17201
Source PR Merged At: 2024-07-12T10:55:24Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/17201
- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2371

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/ingest/ingest-from-gcs.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ FORMAT data_format ENCODE data_encode (
 |gcs.bucket_name |Required. The name of the bucket the data source is stored in. |
 |gcs.credential|Optional. The base64 encoded credential key. If this field is not specified, ADC (application default credentials) will be used. |
 |gcs.service_account|Optional. The service account of the target GCS source. If `gcs.credential` or ADC is not specified, the credentials will be derived from the service account.|
-|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
+|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, you can use file source to read parquet file.|
```
Comment (emile-00) [inline docs/ingest/ingest-from-gcs.md]:
```suggestion
|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, the file source can read parquet files.|
```

File: docs/ingest/ingest-from-s3.md [hunk 2]
```diff
@@ -106,7 +106,7 @@ export const svg = rr.Diagram(
 |s3.bucket_name |Required. The name of the bucket the data source is stored in. |
 |s3.credentials.access|Required. This field indicates the access key ID of AWS. |
 |s3.credentials.secret|Required. This field indicates the secret access key of AWS.|
-|match_pattern| Conditional. This field is used to find object keys in `s3.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
+|match_pattern| Conditional. This field is used to find object keys in `s3.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, you can use file source to read parquet file.|
```
Comment (emile-00) [inline docs/ingest/ingest-from-s3.md]:
```suggestion
|match_pattern| Conditional. This field is used to find object keys in `s3.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, the file source can read parquet files.|
```

File: docs/ingest/ingest-from-gcs.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ FORMAT data_format ENCODE data_encode (
 |gcs.bucket_name |Required. The name of the bucket the data source is stored in. |
 |gcs.credential|Optional. The base64 encoded credential key. If this field is not specified, ADC (application default credentials) will be used. |
 |gcs.service_account|Optional. The service account of the target GCS source. If `gcs.credential` or ADC is not specified, the credentials will be derived from the service account.|
-|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
+|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, the file source can read parquet files.|
```
Comment (wcy-fdu) [inline docs/ingest/ingest-from-gcs.md]:
```suggestion
|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `.parquet`, the file source can read parquet files.|
```

File: docs/ingest/ingest-from-s3.md [hunk 4]
```diff
@@ -106,7 +106,7 @@ export const svg = rr.Diagram(
 |s3.bucket_name |Required. The name of the bucket the data source is stored in. |
 |s3.credentials.access|Required. This field indicates the access key ID of AWS. |
 |s3.credentials.secret|Required. This field indicates the secret access key of AWS.|
-|match_pattern| Conditional. This field is used to find object keys in `s3.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
+|match_pattern| Conditional. This field is used to find object keys in `s3.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. When set to `*.parquet`, the file source can read parquet files.|
```
Comment (wcy-fdu) [inline docs/ingest/ingest-from-s3.md]: ditto. we only need to add `parquet` as suffix.

**Review by wcy-fdu — COMMENTED** [review]
I think we should have documentation to introduce the encoding types (json, csv, parquet) supported by the current file source, instead of just stating it in the match pattern.

**Comment by fuyufjh** [comment]
> I think we should have documentation to introduce the encoding types (json, csv, parquet) supported by the current file source, instead of just stating it in the match patter

+1. There are more places need to be changed in the text. For example, at the beginning of doc [Ingest data from S3 buckets](https://docs.risingwave.com/docs/current/ingest-from-s3/),

> RisingWave supports both CSV and [ndjson](https://ndjson.org/) file formats.

Perhaps you can search for "CSV" to find out all the occurrences.

**Comment by wcy-fdu** [comment]
> > I think we should have documentation to introduce the encoding types (json, csv, parquet) supported by the current file source, instead of just stating it in the match patter
> 
> +1. There are more places need to be changed in the text. For example, at the beginning of doc [Ingest data from S3 buckets](https://docs.risingwave.com/docs/current/ingest-from-s3/),
> 
> > RisingWave supports both CSV and [ndjson](https://ndjson.org/) file formats.
> 
> Perhaps you can search for "CSV" to find out all the occurrences.

Yes, I'm rewriting this part and will commit a pr later.

**Comment by ShanlanLi** [comment]
resolved by https://github.com/risingwavelabs/risingwave-docs/pull/2470#event-14010156319 and https://github.com/risingwavelabs/risingwave-docs/pull/2546

**Pull request closed without merging**